sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.searchStudents.e2e-spec.ts

```diff
@@ -0,0 +1,470 @@
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Submitted,
+      },
+    );
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: "",
+      firstName: "",
+      lastName: student.user.lastName,
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body).toEqual(
+          expect.arrayContaining([
+            {
+              id: student.id,
+              firstName: student.user.firstName,
+              lastName: student.user.lastName,
+              birthDate: student.birthDate,
+              sin: student.sinValidation.sin,
+            },
+          ]),
+        );
+      });
+  });
+
+  it("Should find the student by first name when student has at least one application submitted for the institution.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    let student = await saveFakeStudent(appDataSource);
+    // Refresh student object to get a birthDate with date only
+    student = await studentRepo.findOne({
+      where: { id: student.id },
+      relations: { sinValidation: true },
+    });
+
+    await saveFakeApplicationDisbursements(
+      appDataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Submitted,
+      },
+    );
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: "",
+      firstName: student.user.firstName,
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body).toEqual(
+          expect.arrayContaining([
+            {
+              id: student.id,
+              firstName: student.user.firstName,
+              lastName: student.user.lastName,
+              birthDate: student.birthDate,
+              sin: student.sinValidation.sin,
+            },
+          ]),
+        );
+      });
+  });
+
+  it("Should find the student by sin when student has at least one application submitted for the institution.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    let student = await saveFakeStudent(appDataSource);
```

Comment:
The arrange part seems to be repetitive with same code for many scenarios. we can create a common method within this .ts file ?